src/s3.rs

```diff
@@ -54,6 +59,51 @@ pub fn get_client(shared_config: &aws_config::Config) -> Result<Client> {
     Ok(Client::from_conf(builder.build()))
 }
 
+#[derive(Debug)]
+pub enum ListObjectsError {
+    ListError(SdkError<ListObjectsV2Error>),
+    MissingContents(ListObjectsV2Output),
+}
```

Comment:
Do you intend this to be a public-facing error struct? It might be a good opportunity to try out the `thiserror` crate as a compliment to `anyhow`.